docs/content/en/docs/pipeline-stages/testers/custom.md

```diff
@@ -0,0 +1,109 @@
+
+Multiple custom testers can be defined per test. The Skaffold pipeline will be blocked on the custom test to complete or fail. Skaffold will exit the loop when the first test fails. For ongoing test failures, Skaffold will stop the loop (not continue with the deploy) but will not exit the loop. Skaffold would surface the errors to the user and will keep the dev loop running. Skaffold will continue watching user specified test dependencies and re-trigger the loop whenever it detects another change. 
+
+CustomTester has a configurable timeout option to wait for the command to return. If no timeout is specified, Skaffold will wait until the test command has completed execution. 
+
+### Contract between Skaffold and Custom command
+
+Skaffold will pass in the environment variable `$IMAGE` to the custom command to access the image.
+
+This variable can be set as a flag value input to the custom command `--flag=$IMAGE`.
+
+
+### Configuration
+
+To use a custom command, add a `custom` field to the corresponding test in the `test` section of the `skaffold.yaml`.
+Supported schema for `CustomTest` includes:
+
+{{< schema root="CustomTest" >}}
+
+
+
+### Dependencies for a Custom Test
+
+Users can specify `dependencies` for custom tests so that skaffold knows when to retest during a dev loop. Dependencies can be specified per command. Users could list out directories and/or files to watch per command. If no dependencies are specified, only the script file (if the command is a script file) will be watched as a dependency. Test dependencies cannot trigger rebuild of an image.
+
+Supported schema for `dependencies` includes:
+
+{{< schema root="CustomTestDependencies" >}}
+
+
+#### Paths and Ignore
+
+`Paths` and `Ignore` are arrays used to list dependencies. This can be a glob.
+Any paths in `Ignore` will be ignored by the skaffold file watcher, even if they are also specified in `Paths`.
+`Ignore` will only work in conjunction with `Paths`.
+
+```yaml
+    custom:
+      - command: ./test.sh
+        timeoutSeconds: 60
+        dependencies:
+          paths:
+          -  "*_test.go"
+          -  "test.sh"
+```
+
+#### Command for dependencies
+
+Sometimes users might have a command or a script that can provide the dependencies for a given test. Custom tester can ask Skaffold to execute a custom command, which Skaffold can use to get the dependencies for the test for file watching.
+
+The command *must* return dependencies as a JSON array, otherwise skaffold will error out.
+
+```yaml
+    custom:
+      - command: echo Hello world!!
+        dependencies:
+          command: echo [\"main_test.go\"] 
+```
+
+### File Sync
+
+Syncable files must be included in both the `paths` section of `dependencies`, so that the skaffold file watcher knows to watch them, and the `sync` section, so that skaffold knows to sync them.  
```

Comment:
You can replace `File Sync` with a `Note`; also link the main file sync doc somewhere.

```

>*Note: Adding a file pattern to a test dependency doesn't automatically enable file sync on it.  Refer to the [`file sync`](link) documentation, on how to set that up separately.*

Reply:
Added the note.